### ISSUE TITLE
Apps on Tizen cannot work without native images

### ISSUE THREAD

**Issue by alpencolt** [dotnet/coreclr#11386]
Native images by documentation contain machine code, so if we will remove them JIT compiler should takes code from dll's. But when I try to do it on Tizen-ARM32 Xamarin applications crash (simple apps like "Hello World" works well). `System.Private.CoreLib.ni.dll` is necessary for applications. Is it normal behavior or there is bug?
The reason why I do so is that I need to run application using RyuJIT. Is there the way to ignore native images or compile them using Ryu?

@dotnet/arm32-contrib

**Comment by BruceForstall** [comment]
You should be able to use COMPlus_ZapDisable=1 to prevent using any .ni.dll native image.

**Comment by alpencolt** [comment]
Yes, it helps. Thank you.
But does this behavior OK?
Is there the way to tell crossgen which compiler use? Or I have to modify source code.

**Comment by BruceForstall** [comment]
> But does this behavior OK?

Do you mean: is it OK for apps to fail without having System.Private.CoreLib.ni.dll? The answer should be no.

> Is there the way to tell crossgen which compiler use?

crossgen automatically loads clrjit.dll (on Windows, for example). You might be able to set:
```
COMPlus_JitName=myjit.dll
```
to get it to load `myjit.dll`. Normally, I use something like:
```
COMPlus_AltJitNgen=*
COMPlus_AltJitName=myjit.dll
```
to use the "altjit" mechanism.

**Comment by jkotas** [comment]
Related to https://github.com/dotnet/coreclr/pull/11341

cc @gkhanna79

**Comment by gkhanna79** [comment]
@alpencolt Assuming you are able to generate NI image of `System.Private.CoreLib.dll `as part of the CoreCLR repo build, it will be named as `System.Private.CoreLib.dll` when placed in the binary output folder, while the original IL will be in the `IL` subfolder.

If you were passing the path to `System.Private.CoreLib.ni.dll` in your host, you will now need to pass the path to `System.Private.CoreLib.dll`. Also see the comment at https://github.com/dotnet/coreclr/pull/11341#issuecomment-298968605 as you may need to change the condition at https://github.com/dotnet/coreclr/pull/11341/files/4edbf8deb33ef6cccdbfc113946553c27d24e9b5#diff-1708de55e5220e56e33f7ddb0f465f9cR692 to account for binding as NI image.

@hqueue @hseok-oh @parjong PTAL at this as well.

**Comment by jyoungyun** [comment]
@alpencolt I do not think this issue should be included in 2.0.0 milestone. What about removing 2.0.0 milestone tag?

**Comment by gkhanna79** [comment]
@jyoungyun Let me know if the milestone needs to be changed and I will do the needful.

**Comment by alpencolt** [comment]
@jyoungyun I think so. I've postponed this issue since we have bugs with higher priority.

**Comment by gkhanna79** [comment]
I have moved this to Future.
